sdk/src/azure/core/az_mqtt5_connection_hfsm.c

```diff
@@ -232,6 +279,16 @@ static az_result started(az_event_policy* me, az_event event)
   return ret;
 }
 
+AZ_INLINE int32_t _get_elapsed_time(int64_t current_time_msec, int64_t start_time_msec)
+{
+  int64_t elapsed_time_msec = current_time_msec - start_time_msec;
+  if (elapsed_time_msec < 0 || elapsed_time_msec > INT32_MAX)
+  {
+    az_platform_critical_error();
```

Comment:
Could this scenario happen if the elapsed time is just very long? If so, it might be better to send an error than completely lock up the whole application